src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -92,7 +92,23 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 }
             }
 
-            if (!string.IsNullOrEmpty(ApplicationUri))
+            string externalConfigPath = AppDomain.CurrentDomain.GetData("APP_CONFIG_FILE") as string;
+            if (!string.IsNullOrEmpty(externalConfigPath))
+            {
+                if (Uri.IsWellFormedUriString(externalConfigPath, UriKind.Absolute))
+                {
+                    Uri externalConfigUri = new Uri(externalConfigPath);
```

Comment:
note: https://github.com/dotnet/runtime/pull/35478 always created an URI and checked for `IsFile` but this actually throws when path is relative while this scenario works on full framework

Reply:
This might be worth a comment in the code too.